### PR TITLE
Use *.json files rather than build-log.txt for building cache.

### PR DESCRIPTION
**Pull request by david-mcmahon**
Additionally, add kubernetes-e2e-gke-serial back as a gating job.

### CODE REVIEW

**Comment by ixdy** [comment]
cc @spxtr

File: lib/releaselib.sh [hunk 1]
```diff
@@ -57,11 +57,10 @@ release::update_job_cache () {
     # the cache should be complete after that
     [[ -n ${JOB[$run]} ]] && break
 
-    buildlog=$($GSUTIL cat $logroot/$job/$run/build-log.txt 2>/dev/null)
-    if [[ "$buildlog" =~ Uploading\ build\ result:\ SUCCESS ]] &&
-       [[ "$buildlog" =~ \
-          (build_version=|Found Kubernetes version: )(${VER_REGEX[release]}\.${VER_REGEX[build]}) ]]; then
-      JOB[$run]=${BASH_REMATCH[2]}
+    buildstate=$($GSUTIL cat $logroot/$job/$run/*.json 2>/dev/null)
+    if [[ "$buildstate" =~ \"result\":\ \"SUCCESS\", ]] &&
```
Comment (ixdy) [inline lib/releaselib.sh]: can you use `jq`? or are you trying to stay completely in bash?

Comment (david-mcmahon) [inline lib/releaselib.sh]: I had used jq previously but that also included more gsutil calls and more piping and executing of external tools (jq).
This current implementation makes a single gsutil call and then the rest is regex'ing in bash.  This is the most optimized way of doing it.

Comment (david-mcmahon) [inline lib/releaselib.sh]: SGTM. PTAL.

Comment (ixdy) [inline lib/releaselib.sh]: Still can get away with only one `gsutil` call:
```bash
result=$(echo "${buildstate}" | jq -r '.result|values')
version=$(echo "${buildstate}" | jq -r '.version|values')
```

File: lib/releaselib.sh [hunk 2]
```diff
@@ -57,11 +57,11 @@ release::update_job_cache () {
     # the cache should be complete after that
     [[ -n ${JOB[$run]} ]] && break
 
-    buildlog=$($GSUTIL cat $logroot/$job/$run/build-log.txt 2>/dev/null)
-    if [[ "$buildlog" =~ Uploading\ build\ result:\ SUCCESS ]] &&
-       [[ "$buildlog" =~ \
-          (build_version=|Found Kubernetes version: )(${VER_REGEX[release]}\.${VER_REGEX[build]}) ]]; then
-      JOB[$run]=${BASH_REMATCH[2]}
+    buildstate=$($GSUTIL cat $logroot/$job/$run/*.json 2>/dev/null |\
+                 jq -r '.result,.version')
```
Comment (ixdy) [inline lib/releaselib.sh]: adding `|values` will get rid of the `null`s (though it's not that big a deal)

File: lib/releaselib.sh (hunk 1 above)
Comment (david-mcmahon) [inline lib/releaselib.sh]: I'm probably splitting hairs with my desire to keep external calls to a minimum:
```
$ time echo "$foo" |jq -r '.version,.result'
null
SUCCESS
v1.3.0-alpha.5.216+66149c50fd0055
null

real	0m0.003s
user	0m0.000s
sys	0m0.007s
```

Change coming...

File: lib/releaselib.sh [hunk 3]
```diff
@@ -57,11 +57,11 @@ release::update_job_cache () {
     # the cache should be complete after that
     [[ -n ${JOB[$run]} ]] && break
```
Comment (ixdy) [inline lib/releaselib.sh]: not related to this PR, but I don't see how this ever gets hit - `$run` is always decremented before returning here, so `$JOB[$run]` is always going to be empty.

Comment (david-mcmahon) [inline lib/releaselib.sh]: The JOB dictionary is initially populated with the current cache in $job_file.  
$run starts at the latest build and indexes $JOB up to $cache_limit. 
When ${JOB[$run]} is set, it breaks.

File: lib/releaselib.sh (hunk 1 above)
Comment (david-mcmahon) [inline lib/releaselib.sh]: Thanks for the |values hint.  Didn't know that one! 
PTAL.

**Pull request merged**